vue/components/ui/molecules/search-persistent/save-filter-modal.vue

```diff
@@ -95,25 +121,148 @@ export const SaveFilterModal = {
             return (
                 this.searchData &&
                 this.filterNameData &&
-                Object.keys(this.tenacyValuesData).length > 0
+                Object.keys(this.tenancyCheckboxValuesData).length > 0 &&
+                this.isTenancyFormValid
             );
         },
         filters() {
-            return Object.keys(this.tenacyValuesData).map(tenancy => ({
+            return Object.keys(this.tenancyCheckboxValuesData).map(tenancy => ({
                 name: this.filterNameData,
                 value: this.searchData,
-                tenancy: tenancy,
+                tenancy: this.getSelectedTenancy(tenancy),
                 context: "" // TODO
             }));
+        },
+        isTenancyEmpty() {
+            return this.tenancyItemsData.length === 1;
+        },
+        isTenancyFormValid() {
+            for (const item of this.tenancyItemsData) {
+                if (
+                    this.isTenancyChoiceSelected(item.value) &&
+                    this.getSelectedTenancy(item.value) === null
+                ) {
+                    return false;
+                }
+            }
+
+            return true;
+        },
+        userTenancy() {
+            return { choices: [], selectedValue: "user", selectVisible: false };
+        },
+        builtTenancies() {
+            const builtTenancies = {};
+
+            this.tenancyItems.forEach(item => {
+                if (item.value === "user") builtTenancies[item.value] = this.userTenancy;
+                else {
+                    builtTenancies[item.value] = {

```

Comment:
In the same spirit as https://github.com/ripe-tech/ripe-components-vue/pull/194/files#r368594708 , try to solve this without these deeply nested data structures.